File: src/dataset/typescript/proverb_Exercism-typescript-FilipeCerejo/gemini/GeminiFlash_proverb_v4.ts
export function proverb(...words: string[]): string {
    if (!words.length) {
        return "";
    }

    const lines: string[] = new Array(words.length);
    for (let i = 0; i < words.length - 1; i++) {
        lines[i] = `For want of a ${words[i]} the ${words[i + 1]} was lost.`;
    }
    lines[words.length - 1] = `And all for the want of a ${words[0]}.`;

    return lines.join('\n');
}

File: src/dataset/Java/proverb_exercism-java-blogscot/gemini/GeminiFlash_proverb_v4.java
class Proverb {
  private final String proverb;

  Proverb(String[] words) {
    if (words.length > 0) {
      StringBuilder builder = new StringBuilder();
      String previous = words[0];

      for (int i = 1; i < words.length; i++) {
        String cause = words[i];
        builder.append("For want of a ").append(previous).append(" the ").append(cause).append(" was lost.\n");
        previous = cause;
      }

      builder.append("And all for the want of a ").append(words[0]).append(".");
      proverb = builder.toString();
    } else {
      proverb = "";
    }
  }

  String recite() {
    return proverb;
  }
}